crates/astria-sequencer-relayer/src/finalization_pipeline/head_candidate.rs

```diff
@@ -0,0 +1,81 @@
+use astria_sequencer_types::SequencerBlockData;
+use tendermint::Hash;
+use tendermint_rpc::endpoint::block;
+
+#[derive(Clone, Default, Debug)]
+pub(crate) enum HeadCandidate {
+    ProposedByValidator(SequencerBlockData),
+    CometBft(CometBftHeadCandidate),
+    #[default]
+    Default,
+}
+
+impl TryInto<SequencerBlockData> for HeadCandidate {
+    type Error = ();
+
+    fn try_into(self) -> Result<SequencerBlockData, Self::Error> {
+        use HeadCandidate::*;
+        match self {
+            ProposedByValidator(block) => Ok(block),
+            CometBft(_) => Err(()),
+            _ => unreachable!(),
+        }
+    }
+}
+
+impl HeadCandidate {
+    pub(crate) fn new_from_validator(block: SequencerBlockData) -> Self {
+        Self::ProposedByValidator(block)
+    }
```

Comment:
where is it getting this data from? the relayer only gets data in the form of `block::Response`s as it only gets blocks via polling sequencer(cometbft). might be nicer to have the types for this and `new_from_cometbft` be the same, maybe change `new_from_cometbft` to also take `SequencerBlockData`?